cloud_storage/cloud_storage/overrides/file.py

```diff
@@ -386,8 +412,32 @@ def write_file(file: File) -> File:
 		file.save_file_on_filesystem()
 		return file
 
-	if not file.name:
-		file.autoname()
+	# if a hash-conflict is found, update the existing document with a new file association
+	existing_file_hashes = frappe.get_all(
+		"File", filters={"content_hash": file.content_hash, "file_size": [">", 0]}, pluck="name"
+	)
+
+	if existing_file_hashes:
+		file_doc = frappe.get_doc("File", existing_file_hashes[0])
+		file_doc.associate_files(file.attached_to_doctype, file.attached_to_name)
+		file_doc.save()
+		return
+
+	# if a filename-conflict is found, update the existing document with a new version instead
+	existing_file_names = frappe.get_all(
+		"File", filters={"file_name": file.file_name, "file_size": [">", 0]}, pluck="name"
+	)
+
+	if existing_file_names:
+		file_doc = frappe.get_doc("File", existing_file_names[0])
+		file_doc.update(
+			{"content": file.content, "content_hash": file.content_hash, "content_type": file.content_type}
+		)
+		file_doc.associate_files(file.attached_to_doctype, file.attached_to_name)
+		file = file_doc
+
+	# if not file.name:
+	# 	file.autoname()
```

Comment:
@agritheory, I commented this out, because of the following scenarios:
- If a new file record is being uploaded, the `upload_file` will do a `file.save()` and set a new name
- If a file record exists and is being updated/re-uploaded, this block will never run anyway

Is there a scenario where this autoname is required here?

Reply:
Not that I can think of. I'd rather remove the code entirely than leave bones lying around.

Reply:
Agreed. I wanted to clarify before I removed that block.